### PR TITLE
II-206 Extract II pipeline configuration to file

### PR DESCRIPTION
**Pull request by bkompis**

### CODE REVIEW

**Comment by bkompis** [comment]
Extracted file is loaded as YAML, not evaluated as Groovy, so working with a closure is currently not possible.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/843/ : SUCCESS in 35s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1656/ : SUCCESS in 1m 17s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1917/ : SUCCESS in 1m 27s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/94/ : FAILURE in 37s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/845/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1658/ : SUCCESS in 1m 09s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1919/ : SUCCESS in 1m 10s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/96/ : FAILURE in 23s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/846/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1659/ : SUCCESS in 1m 10s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1920/ : SUCCESS in 1m 20s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/97/ : FAILURE in 46s

**Comment by michalvanco** [comment]
@sadam21 pls take this over as we discussed... I guess that update in pipelines is needed to make it work again, thanks!!!

**Comment by sadam21** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/988/ : SUCCESS in 13s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/239/ : FAILURE in 40s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1001/ : SUCCESS in 17s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/252/ : FAILURE in 48s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1003/ : SUCCESS in 10s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/254/ : FAILURE in 42s

**Comment by bkompis** [comment]
Blocked by https://github.com/gooddata/pipelines-shared-libs/pull/162 - needs support in pipelines.

**Comment by bkompis** [comment]
the argument closure may not use any Pipeline steps, only Groovy. Retest after pipelines are updated.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1006/ : SUCCESS in 10s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/257/ : FAILURE in 40s

**Comment by michalvanco** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1032/ : SUCCESS in 20s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/283/ : FAILURE in 1m 02s

File: .gdc-ii-config.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+microservices:
+  lcm-bricks:
+    docker:
+      dockerfile: './Dockerfile'
+      arguments:
+        BRICKS_VERSION: '{ workspacePath -> return new File("${workspacePath}/gooddata-ruby/VERSION").text.trim() }'
```
Comment (michalvanco) [inline .gdc-ii-config.yaml]: drop `gooddata-ruby` from the value

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1036/ : SUCCESS in 8s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/287/ : FAILURE in 51s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1038/ : SUCCESS in 26s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/289/ : FAILURE in 48s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1039/ : SUCCESS in 50s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/290/ : FAILURE in 39s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1040/ : SUCCESS in 10s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/291/ : FAILURE in 42s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1041/ : SUCCESS in 8s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/292/ : FAILURE in 56s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1043/ : SUCCESS in 35s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/294/ : SUCCESS in 8m 17s

**Comment by bkompis** [comment]
The previous build is incorrect - it uses a hard-coded value for BRICKS_VERSION.

**Comment by panjan** [comment]
@bkompis, what is the status of this PR? Is it still valid? Thanks.

**Comment by bkompis** [comment]
It seems the solution from https://github.com/gooddata/pipelines-shared-libs/pull/162 is not enough to enable this - I'll have to look into it again. It's possible another change to `pipelines-shared-libs` will be needed.

The version currently present here is invalid, please keep the PR open - I'll try to fix it asap, as it's been a problem for a long time already.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/1109/ : Tests on opensource repositories should be triggered by users in `gooddata` organization in 15s
- gooddata-ruby-build-pipeline : SKIPPED

**Comment by bkompis** [comment]
Loading of files has finally been fixed (see II-328), closing this PR in favour of #1439.

**Pull request closed without merging**